### PR TITLE
Bugfix - Scale new_app separately from killing old_app

### PR DESCRIPTION
**Pull request by JamesAwesome**
@brndnmtthws this should address both #190 and #191 Normally I would just work on the original PR with you, but I saw a chance to kill two birds with one PR.

There is one difference here, and that's that i simply wait for apps to be fully drained (no qcur and no scur) before i delete their tasks.

### CODE REVIEW

**Comment by brndnmtthws** [comment]
> There is one difference here, and that's that i simply wait for apps to be fully drained (no qcur and no scur) before i delete their tasks.

The thing is, the timeout is there to handle the case of "what happens if my backend never drains?" i.e., maybe there are very long lived connections, and you eventually (say, after 5 minutes) want to cut them off so the deploy can proceed.

Minus that detail, the PR looks good.

By the way, checking qcur and scur is kind of pointless, but it's okay to do it anyway. The only indicator that there are still connections to the draining backends is the number of HAProxy PIDs.

File: bluegreen_deploy.py [hunk 1]
```diff
@@ -267,31 +264,45 @@ def swap_bluegreen_apps(args, new_app, old_app, timestamp):
         if waiting_for_drained_listeners(listeners):
             continue
 
-        drained_task_ids = \
-            find_drained_task_ids(old_app, listeners, haproxy_count)
+        return find_drained_task_ids(old_app, listeners, haproxy_count)
+
+    logger.info('Timed out waiting for drained tasks, Continuing...')
+    return []
```
Comment (brndnmtthws) [inline bluegreen_deploy.py]: As per my other comment, we should just return all the backends marked for draining at this point.

**Comment by JamesAwesome** [comment]
Cool, got it! 😄 Will most likely get this PR updated Monday afternoon-ish 👍

**Comment by JamesAwesome** [comment]
Okay, so the way this is setup now is that if we time out during `find_tasks_to_kill` we return task ids for any listeners which are marked 'MAINT' (regardless of scur/qcur) across all haproxy backends. However, if we find any listeners with fully drained tasks (no scur/qcur) prior to the timeout we kill those tasks first.

**Comment by brndnmtthws** [comment]
This is looking right, thank you.

**Pull request merged**